src/serial/serial_buffer.c

```diff
@@ -0,0 +1,173 @@
+/*
+ * Race Capture Pro Firmware
+ *
+ * Copyright (C) 2015 Autosport Labs
+ *
+ * This file is part of the Race Capture Pro fimrware suite
+ *
+ * This is free software: you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ *
+ * See the GNU General Public License for more details. You should
+ * have received a copy of the GNU General Public License along with
+ * this code. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "mem_mang.h"
+#include "printk.h"
+#include "serial.h"
+#include "serial_buffer.h"
+#include "stdutil.h"
+#include "string.h"
+#include "taskUtil.h"
+
+#include <stdarg.h>
+#include <stdbool.h>
+#include <stdio.h>
+
+static void ctrl_char_strip(char *data)
+{
+        /* Strip all control characters from our messages */
+        for(; *data >= 32; ++data);
+        *data = 0;
+}
+
+bool serial_buffer_create(struct serial_buffer *sb,
```

Comment:
We have quite a bit of RAM available in the CCR block of the STM32 (the 64K region). if we define these buffers statically, we can conserve the 128K RAM block for Lua and other things that need dynamic RAM. Thoughts?